opensourceleg/actuators/moteus.py

```diff
@@ -96,97 +145,202 @@ def set_position(self, value: Union[float, int]):
+        )
+
+    def _entry(self) -> None:
+        LOGGER.debug(msg=f"[MoteusControlMode] Entering {self.name} mode.")
 
         if self.actuator is None:
             raise ActuatorIsNoneException(mode=self.name)
 
         if not self.has_gains:
             self.set_gains()
 
-        self.set_current(value=0)
+        self.set_position(value=0)
 
     def _exit(self) -> None:
-
-        LOGGER.debug(msg=f"[DephyControlMode] Exiting {self.name} mode.")
+        LOGGER.debug(msg=f"[MoteusControlMode] Exiting {self.name} mode.")
 
         # Is this necessary? This was a required step for older flexsea but not sure if it is needed anymore
         
-        time.sleep(1 / self._actuator.frequency)
+        time.sleep(0.1)
 
-    def set_gains(self, gains: ControlGains = DEFAULT_CURRENT_GAINS) -> None:
+    async def set_gains(
+        self,
+        gains: ControlGains = DEFAULT_POSITION_GAINS,
+    ) -> None:
         super().set_gains(gains)
-        # self.actuator.set_gains(
-        #     kp=int(gains.kp),
-        #     ki=int(gains.ki),
-        #     kd=0,
-        #     k=0,
-        #     b=0,
-        #     ff=int(gains.ff),
-        # )
+        await self._actuator._stream.command(
+            f'conf set servo.pid_position.kp {self._gains.kp}'.encode('utf8')
+        )
+        await self._actuator._stream.command(
+            f'conf set servo.pid_position.ki {self._gains.ki}'.encode('utf8')
+        )
+        await self._actuator._stream.command(
+            f'conf set servo.pid_position.kd {self._gains.kd}'.encode('utf8')
+        )
 
-    def set_current(self, value: Union[float, int]):
-        self._actuator._command = self._actuator.make_current( 
-            d_A = value, 
-            q_A = 0, 
+    def set_position(self, value: Union[float, int]):
+        print(value)
+        self.actuator._command = self.actuator.make_position(
+            # kp_scale = self._gains.kp, 
+            # kd_scale = self._gains.kd,
+            # ilimit_scale = self._gains.ki,
+            position = float((value) / (2 * np.pi)), # in revolutions
             query = True,
+            watchdog_timeout = math.nan,
+        )
+    def set_current(self, value: Union[float, int]):
+        raise ControlModeException(
+            tag=self.actuator.tag,
+            attribute=str(ControlModesMapping.CURRENT),
+            mode=self.name,
         )
 
+    def set_velocity(self, value: Union[float, int]):
+        raise ControlModeException(
+            tag=self.actuator.tag,
+            attribute=str(ControlModesMapping.VELOCITY),
+            mode=self.name,
+        )
+    
     def set_voltage(self, value: Union[float, int]):
         raise ControlModeException(
             tag=self.actuator.tag,
             attribute=str(ControlModesMapping.VOLTAGE),
             mode=self.name,
         )
-
-    def set_position(self, value: Union[float, int]):
+    
+    def set_stop(self):
```

Comment:
We could try to remove these stop methods. `set_stop` inside position control mode is not intuitive.